includes/abstracts/abstract-wc-rest-terms-controller.php

```diff
@@ -380,20 +380,8 @@ public function create_item( $request ) {
 		if ( isset( $request['slug'] ) ) {
 			$args['slug'] = $request['slug'];
 		}
-
 		if ( isset( $request['parent'] ) ) {
-			if ( ! is_taxonomy_hierarchical( $taxonomy ) ) {
```

Comment:
We still need to check if is hierarchical first, WordPress by default does not check it while using `wp_insert_term` or `wp_update_term`.